course_info/static/course_info/js/controllers/DetailsController.js

```diff
@@ -72,35 +77,39 @@
         };
 
         dc.dissociateSite = function(siteListIndex) {
-            // todo: siteId is not currently available from backend; once added we can use that to call backend (instead of siteListIndex) and remove from course instance object
-            $scope.confirmDissociateSiteModalInstance = $uibModal.open({
+            dc.confirmDissociateSiteModalInstance = $uibModal.open({
                 animation: true,
                 templateUrl: 'partials/dissociate-site-confirmation.html',
-                controller: function ($scope, $uibModalInstance, siteURL) {
-                    $scope.siteURL = siteURL;
+                controller: function ($scope, $uibModalInstance, siteURL, site_map_id) {
+                    dc.siteURL = siteURL;
+                    dc.site_map_id = site_map_id;
                 },
                 resolve: {
                     siteURL: function () {
                         return dc.courseInstance.sites[siteListIndex].course_site_url;
+                    },
+                    site_map_id: function() {
+                        return dc.courseInstance.sites[siteListIndex].site_map_id;
                     }
                 }
             });
+            dc.confirmDissociateSiteModalInstance.result.then(
+                function modalConfirmed() {
+                    var delete_course_site_url = djangoUrl.reverse(dc.apiProxy,
+                        [dc.apiBase + dc.courseInstance.course_instance_id + '/sites/' + dc.courseInstance.sites[siteListIndex].site_map_id + '/']);
+                    $http.delete(delete_course_site_url)
+                        .then(function(response){
+                            dc.dissociateSiteInProgressIndex = siteListIndex;
```

Comment:
Yes, because the ['in progress' indicator on the template](https://github.com/Harvard-University-iCommons/canvas_account_admin_tools/blob/task/elliottyates/tlt-2570/course_details_add_remove_course_sites/course_info/templates/course_info/partials/details.html#L266) is linked to this -- but if we only set it here, then it won't ever show (or it will flicker very quickly) as here we're already finished with the operation. So I think it should be before line 100, no?